lib/ASRouter.jsm

```diff
@@ -46,6 +81,50 @@ const MAX_MESSAGE_LIFETIME_CAP = 100;
 const LOCAL_MESSAGE_PROVIDERS = {OnboardingMessageProvider, CFRMessageProvider, SnippetsTestMessageProvider};
 const STARTPAGE_VERSION = "6";
 
+/**
+ * chooseBranch - choose from a list of optional branches and ratios
+ * @param branches {Array<[any, number?]>} A list of branches, where branch[0] is any item and branch[1] is the ratio
+ * @param seed {string} A unique seed for the randomizer
+ */
+async function chooseBranch(branches, seed) {
+  const ratios = branches.map(([item, ratio]) => ((typeof ratio !== "undefined") ? ratio : 1));
+  return branches[await Sampling.ratioSample(seed, ratios)][0];
+}
+
+async function generateTrailheadBranches() {
+  let experiment = "";
+  let interupt;
+  let triplet;
+
+  // We can use a pref to override and bypass the experiment process
+  const overrideValue = Services.prefs.getStringPref(TRAILHEAD_OVERRIDE_PREF, "");
+  if (overrideValue) {
+    [interupt, triplet] = overrideValue.split("-");
+    return {experiment, interupt, triplet: triplet || ""};
+  }
+
+  const locale = Services.locale.appLocaleAsLangTag && Services.locale.appLocaleAsLangTag.substr(0, 2);
```

Comment:
The locale code can be 3 letters although in our usage, there's none that are enX-, frX- or deX- anyway. Also `appLocaleAsLangTag` is guaranteed to be a string (so even if it's somehow empty string, substr will work)

Reply:
I'm updating this to be the full locale, since we don't need sub-buckets for locales